cpp/lib/ome/xerces/dom/Document.h

```diff
@@ -177,6 +178,60 @@ namespace ome
         {
           return Element((*this)->getDocumentElement(), false);
         }
+
+        /**
+         * Get child elements with a given tag name.
+         *
+         * @param name the element name to use.
+         * @returns the child nodes (if any).
+         */
+        NodeList
+        getElementsByTagName(const std::string& name)
+        {
+          return (*this)->getElementsByTagName(String(name));
+        }
+      };
+
+      /**
+       * Parameters controlling DOM writing.
+       *
+       * The DOMSerializer provides for some control over the process
+       * via DOMConfiguration.  They are settable here to allow their
+       * use at a high level without the need to have access to the
+       * internals of the Xerces-C writing process.  Simply create an
+       * instance of this class, adjust the parameters as needed, and
+       * then pass to a method call which uses WriteParameters as an
+       * optional argument.
+       *
+       * If more precise control of the process is required, use the
+       * Xerces-C classes directly.  These are simply a convienience
+       * for the common case and will not suit every situation.
+       */
+      struct ParseParameters
+      {
+        /// Validation scheme.
+        xercesc::XercesDOMParser::ValSchemes validationScheme;
+        /// Use namespaces?
+        bool doNamespaces;
+        /// Use schemas?
+        bool doSchema;
+        /// Handle multiple imports?
+        bool handleMultipleImports;
+        /// Do full checking during validation?
+        bool validationSchemaFullChecking;
```

Comment:
It maps directly onto the Xerces API, so while I don't think it's really that obvious myself, it's directly matching the Xerces behaviour.  The entire ome-xerces wrapper is a thin layer over the Xerces-C API, so it's only intended to be used in combination with the Xerces API reference.